packages/airnode-deployer/src/infrastructure/index.ts

```diff
@@ -511,6 +512,79 @@ export type Deployment = {
   versions?: DeploymentVersion[];
 };
 
+// If deploymentId is provided tries fetch only that one deployment and finishes early when found
+async function fetchDeployments(cloudProvider: CloudProvider['type'], deploymentId?: string) {
```

Comment:
Nit: I think we will be using type `CloudProvider['type']` so consider defining it as `type CloudProviderType = CloudProvider['type']`

Reply:
I'm currently replacing it with a full `CloudProvider` type at some places so I'll see whether it makes sense.